processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryQueryToolChest.java

```diff
@@ -181,9 +181,11 @@ public SegmentAnalysis apply(SegmentAnalysis arg1, SegmentAnalysis arg2)
       public byte[] computeCacheKey(SegmentMetadataQuery query)
       {
         byte[] includerBytes = query.getToInclude().getCacheKey();
-        return ByteBuffer.allocate(1 + includerBytes.length)
+        byte[] analysisTypesBytes = query.getAnalysisTypesCacheKey();
```

Comment:
Shall I reintroduce the prefix byte then? Can give it a value of something like 0xA0 so it wouldn't be confused with part of a valid UTF-8 column name.